doc/source/development/extending.rst

```diff
@@ -134,6 +134,47 @@ by some other storage type, like Python lists.
 See the `extension array source`_ for the interface definition. The docstrings
 and comments contain guidance for properly implementing the interface.
 
+:class:`~pandas.api.extensions.NDArrayBackedExtensionArray`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For ExtensionArrays backed by a single NumPy array, the
+:class:`~pandas.api.extensions.NDArrayBackedExtensionArray` class can save you
+some effort. It contains a private property ``_ndarray`` with the backing NumPy
+array and implements the extension array interface. Implement the ``_box_func``
+method to convert from array values to the type you wish to expose to users.
+Implement the ``_validate_scalar`` method to convert from an object to a value
+which can be stored in the NumPy array.
+
+.. code-block:: python
+
+    class CustomArray(NDArrayBackedExtensionArray):
+        def __init__(self, values):
+            backing_array_dtype = "int64"
+            super().__init__(values=values, dtype=backing_array_dtype)
```

Comment:
the dtype passed here should be an ExtensionDtype object (we probably ought to do some validation on that)

Reply:
Really? We're just passing in the numpy dtype here:

https://github.com/googleapis/python-db-dtypes-pandas/blob/e9d41d17b5d6a7d83c46e2497feb8e314545adcb/db_dtypes/core.py#L53

I'll look into this some more